src/opencl_common.c

```diff
@@ -2829,6 +2883,296 @@ int opencl_calc_min_kpc(size_t lws, size_t gws, int v_width)
 	return gws * v_width;
 }
 
+static void buf_list_init(struct opencl_buffers **list)
+{
+	*list = mem_alloc(sizeof(struct opencl_buffers));
+	(*list)->tail = (*list)->head = NULL;
+	(*list)->count = 0;
+}
+
+static void buf_list_add_link(struct opencl_buffers *list, struct buffer_opencl *entry)
+{
+	entry->next = NULL;
+
+	if (list->tail)
+		list->tail = list->tail->next = entry;
+	else
+		list->tail = list->head = entry;
+
+	list->count++;
+}
+
+static void buf_list_add(struct opencl_buffers *list, cl_mem *data, void **host_ptr)
+{
+	struct buffer_opencl *entry;
+
+	entry = mem_alloc(sizeof(struct buffer_opencl) + sizeof(data));
+	entry->data = *(data);
+	entry->variable = data;
+	entry->host_ptr = host_ptr;
+
+	buf_list_add_link(list, entry);
+}
+
+static void buf_list_remove(struct opencl_buffers *list, cl_mem data)
+{
+	struct buffer_opencl *current, *prev = NULL;
+
+	for (current = list->head; current; current = current->next) {
+		if (current->data == data) {
```

Comment:
Why do we choose to find the thing to remove by value?

Reply:
To allow a `release_buf(pointer)` in the future.

Reply:
OK. We might or might not need that. But anyway, why search by value?